internal/checker/cloudformation/rules/iam/verify_policy_groups_roles.rego

```diff
@@ -0,0 +1,48 @@
+# METADATA
+# title: "Verify IAM policies attached only to groups or roles"
+# description: "IAM users, groups, and roles lack access to AWS resources by default. Privileges are granted through IAM policies, which streamline access management by assigning privileges at the group or role level. This approach minimizes complexity, especially with a growing user base, and helps prevent unintended privilege escalation. We advise applying IAM policies directly to groups and roles rather than individual users."
+# scope: package
+# related_resources:
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-policy.html
+# custom:
+#   id: CB_CFT_31
+#   severity: LOW
+package lib.cloudformation.CB_CFT_31
+import future.keywords.in
+
+resource := "AWS::IAM::Policy"
+
+is_valid {
+	some resources in input.Resources
+	resources.Type == resource
+}
+
```

Comment:
should pass if the users is not present
try this instead

pass[properties] {
	is_valid
	some resources in input.Resources
	properties := resources.Properties
	not "Users" in object.keys(properties)
}